src/world/action/land.c

```diff
@@ -1,9 +1,220 @@
+        sfx_play_sound_at_player(0x148, 0);
+
+        if (!(currentCollisionStatus->currentFloor & 0x4000)) {
+            phys_adjust_cam_on_landing();
+        }
+
+        currentCollisionStatus->lastTouchedFloor = -1;
+        playerStatus->animFlags &= ~PLAYER_STATUS_ANIM_FLAGS_40000;
+        currentCameras->moveFlags = currentCameras->moveFlags & 0xFFFB;
+    }
+    playerStatus->fallState++;
+    playerStatus->currentSpeed *= 0.6f;
+
+    player_input_to_move_vector(&inputMoveAngle, &inputMoveMagnitude);
+    jumpInputCheck = check_input_jump();
+
+    if ((jumpInputCheck != 0) || (jumpInputCheck < playerStatus->fallState)) {
+        if (inputMoveMagnitude == 0.0f) {
+            set_action_state(0);
+            return;
+        }
+        if (inputMoveMagnitude != 0.0f) {
+            playerStatus->targetYaw = inputMoveAngle;
+        }
+        update_locomotion_state();
+    }
+}
+
+void func_802B61C0_E24AE0(void) {
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+
+    if ((playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_USING_PEACH_PHYSICS)) {
+        func_802B644C_E24D6C();
+        return;
+    }
+
+    if (playerStatus->flags & PLAYER_STATUS_FLAGS_ACTION_STATE_CHANGED) {
+         playerStatus->flags = playerStatus->flags & ~(
+            PLAYER_STATUS_FLAGS_ACTION_STATE_CHANGED | 
+            PLAYER_STATUS_FLAGS_800000 | 
+            PLAYER_STATUS_FLAGS_80000 | 
+            PLAYER_STATUS_FLAGS_8 |
+            PLAYER_STATUS_FLAGS_FALLING | 
+            PLAYER_STATUS_FLAGS_JUMPING
+        );
+        playerStatus->fallState = 0;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->decorationPos[0] = playerStatus->position.x;
+        playerStatus->decorationPos[1] = playerStatus->position.z;
+
+        if (!(currentCollisionStatus->currentFloor & 0x4000)) {
+            phys_adjust_cam_on_landing();
+        }
+
+        currentCollisionStatus->lastTouchedFloor = -1;
+    }
+
+    playerStatus->fallState = (u8) playerStatus->fallState + 1;
+    playerStatus->currentSpeed *= 0.6f;
+
+    player_input_to_move_vector(&inputMoveAngle, &inputMoveMagnitude);
+    check_input_jump();
+
+    if (inputMoveMagnitude != 0.0f) {
+        playerStatus->targetYaw = inputMoveAngle;
+    }
+    
+    update_locomotion_state();
+}
+
+void func_802B62CC_E24BEC(void) {
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+    s32 playerFlagsModified;
+    s32 squaredStick0;
+    s32 squaredStick1;
+
+    if (playerStatus->flags & PLAYER_STATUS_FLAGS_ACTION_STATE_CHANGED) {
+        playerFlagsModified = playerStatus->flags & ~PLAYER_STATUS_FLAGS_ACTION_STATE_CHANGED;
+        playerStatus->flags = playerFlagsModified;
```

Comment:
playerFlagsModified doesn't need to exist and this can just be playerStatus->flags &= ~xyz. below you cna do &= as well